### PR TITLE
Empty trie

### PR DESCRIPTION
**Pull request by praetoriansentry**
Adding the witness fixes from @wborgeaud  and minor documentation notes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,14 +2,132 @@
 
 Similar to [`eth-proof`](https://github.com/wborgeaud/eth-proof) but for transaction proofs.
 
+## Quick Start
+
+There two ways to run this prover. The simplest way to get started is
```
Comment (Nashtare) [inline README.md:7]: nit:
```suggestion
There are two ways to run this prover. The simplest way to get started is
```

File: README.md [hunk 2]
```diff
@@ -2,14 +2,132 @@
...
+```bash
+paladin-leader rpc -u $RPC_URL -t 0x2f0faea6778845b02f9faf84e7e911ef12c287ce7deb924c5925f3626c77906e > 0x2f0faea6778845b02f9faf84e7e911ef12c287ce7deb924c5925f3626c77906e.json
+```
+
+You'll need access to an Ethereum RPC in order to run this
+command. The input argument is a transaction hash and in particular it
+is the *last* transaction has in the block.
+
+Once you've successfully generated a witness, you're ready to start
+proving either with the `in-memory` run time or the `amqp` runtime.
```
Comment (Nashtare) [inline README.md:44]: nit:
```suggestion
proving either with the `in-memory` runtime or the `amqp` runtime.
```

File: README.md [hunk 3]
```diff
@@ -2,14 +2,132 @@
...
+
+### In Memory Proving
+
+Running the prover with the `in-memory` setup requires no setup. You
```
Comment (Nashtare) [inline README.md:48]: nit, to avoid repetition
```suggestion
Running the prover with the `in-memory` flag requires no setup. You
```

**Review by Nashtare — APPROVED** [review]
couple nits just to make peace with my OCD, but LGTM :)
Probably not worth adding a TODO to halve the number of RPCs for block_hashes fetching, that's not really a bottleneck anymore compared to the state trie.

**Review by wborgeaud — APPROVED** [review]
Thanks 👍

**Pull request merged**